source/extensions/filters/http/compressor/compressor_filter.h

```diff
@@ -160,6 +160,19 @@ class CompressorFilterConfig {
 };
 using CompressorFilterConfigSharedPtr = std::shared_ptr<CompressorFilterConfig>;
 
+class CompressorPerRouteFilterConfig : public Router::RouteSpecificFilterConfig {
+public:
+  CompressorPerRouteFilterConfig(
+      const envoy::extensions::filters::http::compressor::v3::CompressorPerRoute& config)
+      : config_(config) {}
+  const envoy::extensions::filters::http::compressor::v3::CompressorPerRoute& config() const {
+    return config_;
+  }
+
+private:
+  const envoy::extensions::filters::http::compressor::v3::CompressorPerRoute config_;
```

Comment:
Typically these classes parse the config, validate its correctness, and keep any relevant info in c++ objects. In this case there might not be much of a difference to keep the original proto.

The added benefit of config validation is that it can throw an exception if the config is invalid, resulting in the config rejection. Not sure if it's relevant to your change.

Reply:
The validation is done in `CommonFactoryBase::createRouteSpecificFilterConfig`, as is the standard pattern. I added a test to config_test.cc to confirm that.

I don't see much value in having this class store a non-proto representation of CompressorPerRoute, because it would just mirror the proto structure, but with everything wrapped in std::optional. It would increase code size and maintenance burden, and mostly decrease readability.

I'll do it if you think the benefits outweigh these costs.

Reply:
Converted per-route proto to optional<bool>.  If more per-route overrides are added, the converted form of the proto will get more optional members, but global consistency may well outweigh the local code simplicity cost.